src/main/java/de/tum/in/www1/artemis/service/exam/ExamImportService.java

```diff
@@ -138,34 +134,73 @@ public List<ExerciseGroup> importExerciseGroupsWithExercisesToExistingExam(List<
+     * Checks that all programming exercises of the given exercise group have a unique title and short name.
+     * Additionally, checks if an exercise with the same project key or name already exists on the VCS / CI.
      *
-     * @param exerciseGroupList the list of all exercises (not only programming) to be checked
+     * @param exerciseGroups  the list of all exercises (not only programming) to be checked
+     * @param courseShortName the short name of the course the exercise will be imported into
      * @throws ExamConfigurationException in case one or more programming exercise project keys are not unique
      */
-    private void preCheckProgrammingExercisesForShortNameUniqueness(List<ExerciseGroup> exerciseGroupList, String targetCourseShortName) {
-        // Flag to determine, if a programming exercise with an invalid shortName was found
-        AtomicInteger numberOfInvalidProgrammingExercises = new AtomicInteger(0);
-        // Iterate over all exercises
-        exerciseGroupList.forEach(exerciseGroup -> {
-            exerciseGroup.getExercises().forEach(exercise -> {
+    private void preCheckProgrammingExercisesForTitleAndShortNameUniqueness(List<ExerciseGroup> exerciseGroups, String courseShortName) {
+        // check for duplicated titles
+        List<String> titles = exerciseGroups.stream().flatMap(group -> group.getExercises().stream()).filter(ex -> ex instanceof ProgrammingExercise).map(BaseExercise::getTitle)
+                .toList();
+        Set<String> uniqueTitles = new HashSet<>(titles);
+
+        if (titles.size() != uniqueTitles.size()) {
+            exerciseGroups.forEach(exerciseGroup -> exerciseGroup.getExercises().forEach(exercise -> {
                 if (exercise.getExerciseType() == ExerciseType.PROGRAMMING) {
-                    // Method to check, if the project already exists.
-                    boolean invalidShortName = programmingExerciseService.preCheckProjectExistsOnVCSOrCI((ProgrammingExercise) exercise, targetCourseShortName);
-                    if (invalidShortName) {
-                        // If the project already exists and thus the short name isn't valid, it is removed
-                        exercise.setShortName("");
+                    if (!uniqueTitles.contains(exercise.getTitle())) {
                         exercise.setTitle("");
-                        numberOfInvalidProgrammingExercises.getAndIncrement();
+                    }
+                    else {
+                        uniqueTitles.remove(exercise.getTitle());
                     }
                 }
-            });
+            }));
+            throw new ExamConfigurationException(exerciseGroups, 0, "duplicatedProgrammingExerciseTitle");
+        }
+
+        // check for duplicated short names
+        List<String> shortNames = exerciseGroups.stream().flatMap(group -> group.getExercises().stream()).filter(ex -> ex instanceof ProgrammingExercise)
```

Comment:
It would be great if the logic to check duplicated titles & names is moved into a method that can be used for both.